### ISSUE TITLE
CompletionException: java.io.IOException: can't read from InputStream

### ISSUE THREAD

**Issue by dotStart** [LordAkkarin/Beacon#23]
Application Crash when the router simply terminates the connection without a proper response.

**Reproduction Steps**
??? - Illegal router response

**Expected Behavior**
An error dialog should be displayed instead.

**Environment:**
 - OS: Windows 10
 - Application Version: 2.4.0

**Additional Context**
```
Message: CompletionException: java.io.IOException: can't read from InputStream
  java.util.concurrent.CompletableFuture.reportJoin in CompletableFuture.java::412
  java.util.concurrent.CompletableFuture.join in CompletableFuture.java::2044
   
  tv.dotstart.beacon.forwarding.ActionExtensionsKt.invoke in ActionExtensions.kt::48
  tv.dotstart.beacon.forwarding.ActionExtensionsKt.invoke$default in ActionExtensions.kt::33
  tv.dotstart.beacon.forwarding.InternetGatewayDevice.getExternalAddress in InternetGatewayDevice.kt::62
  tv.dotstart.beacon.forwarding.PortExposureProvider.load in PortExposureProvider.kt::66
  tv.dotstart.beacon.preload.Preloader$invoke$1.invoke in Preloader.kt::88
  tv.dotstart.beacon.preload.Preloader$invoke$1.invoke in Preloader.kt::35
  kotlin.concurrent.ThreadsKt$thread$thread$1.run in Thread.kt::30
Message: IOException: can't read from InputStream
  net.mm2d.upnp.internal.message.HttpMessageDelegate$Companion.readLine in HttpMessageDelegate.kt::302
  net.mm2d.upnp.internal.message.HttpMessageDelegate$Companion.access$readLine in HttpMessageDelegate.kt::269
  net.mm2d.upnp.internal.message.HttpMessageDelegate.readStartLine in HttpMessageDelegate.kt::216
  net.mm2d.upnp.internal.message.HttpMessageDelegate.readData in HttpMessageDelegate.kt::204
  net.mm2d.upnp.HttpResponse.readData in HttpResponse.kt::-1
  net.mm2d.upnp.HttpResponse$Companion.create in HttpResponse.kt::136
  net.mm2d.upnp.HttpClient.writeAndRead in HttpClient.kt::144
  net.mm2d.upnp.HttpClient.doRequest in HttpClient.kt::125
  net.mm2d.upnp.HttpClient.post in HttpClient.kt::104
  net.mm2d.upnp.HttpClient.post in HttpClient.kt::88
  net.mm2d.upnp.internal.impl.ActionInvokeDelegate.invoke in ActionInvokeDelegate.kt::111
  net.mm2d.upnp.internal.impl.ActionInvokeDelegate.invoke in ActionInvokeDelegate.kt::92
  net.mm2d.upnp.internal.impl.ActionInvokeDelegate.invoke in ActionInvokeDelegate.kt::49
  net.mm2d.upnp.internal.impl.ActionImpl.invokeCustomSync in ActionImpl.kt::49
  net.mm2d.upnp.internal.impl.ActionImpl$invokeInner$1.invoke in ActionImpl.kt::63
  net.mm2d.upnp.internal.impl.ActionImpl$invokeInner$1.invoke in ActionImpl.kt::22
  net.mm2d.upnp.internal.thread.ExecuteFunctionKt$toRunnable$1.run in ExecuteFunction.kt::19
  java.util.concurrent.ThreadPoolExecutor.runWorker in ThreadPoolExecutor.java::1128
  java.util.concurrent.ThreadPoolExecutor$Worker.run in ThreadPoolExecutor.java::628
  java.lang.Thread.run in Thread.java::834
```

**Comment by dotStart** [comment]
Raygun report https://app.raygun.com/crashreporting/27ce4ol/errors/73714501549 has more details about the error

**Comment by dotStart** [comment]
The issue has already been addressed in 3.0.
